src/sources/datadog/agent.rs

```diff
@@ -322,10 +502,86 @@ fn decode(header: &Option<String>, mut body: Bytes) -> Result<Bytes, ErrorMessag
             }
         }
     }
-
     Ok(body)
 }
 
+fn into_vector_metric(dd_metric: DatadogSeriesMetric, api_key: Option<Arc<str>>) -> Vec<Event> {
+    let mut tags = BTreeMap::<String, String>::new();
+    for tag in dd_metric.tags.clone().unwrap_or_default() {
+        let kv = tag.split_once(":").unwrap_or((&tag, ""));
+        tags.insert(kv.0.trim().into(), kv.1.trim().into());
+    }
+    dd_metric
+        .host
+        .as_ref()
+        .and_then(|host| tags.insert(log_schema().host_key().to_owned(), host.to_owned()));
+    dd_metric
+        .source_type_name
+        .as_ref()
+        .and_then(|source| tags.insert("source_type_name".into(), source.to_owned()));
+    dd_metric
+        .device
+        .as_ref()
+        .and_then(|dev| tags.insert("device".into(), dev.to_owned()));
```

Comment:
Since this function owns the `dd_metric`, these should be doable without the `as_ref`, one way or another, which avoids cloning them.

Reply:
Indeed, just dropping `as_ref` works smoothly, thanks !